### ISSUE TITLE
vsce verify-pat does not seem to use `publisher` from `package.json`

### ISSUE THREAD

**Issue by scarf005** [felipecrs/semantic-release-vsce#341]
# Log
- [`.releaserc.yml`](https://github.com/scarf005/Cataclysm-BN-tools/blob/main/.releaserc.yml)
- workflow file: https://github.com/scarf005/Cataclysm-BN-tools/actions/runs/3256483667/workflow
- excerpt from https://github.com/scarf005/Cataclysm-BN-tools/actions/runs/3256483667/jobs/5347034331

```sh
pnpx vsce verify-pat scarf --pat $VSCE_PAT
The Personal Access Token verification succeeded for the publisher 'scarf'.

pnpx semantic-release
Error: Command failed with ENOENT: vsce verify-pat
spawn vsce ENOENT
AggregateError: 
    SemanticReleaseError: Invalid vsce token. Additional information:
    Error: Command failed with ENOENT: vsce verify-pat
    spawn vsce ENOENT
```
I've correctly set up `publisher` as `scarf` in my [package.json](https://github.com/scarf005/Cataclysm-BN-tools/blob/057b6c591c788daa71b54ec439c0a7b3622950bd/package.json), however it fails with an error.

**Comment by felipecrs** [comment]
Can you try running vsce verify-pat in your command line to check the validity of your token?

**Comment by felipecrs** [comment]
Oh, I think you did already. Thanks.

**Comment by felipecrs** [comment]
This may be an incompatibility with pnpm. Can you try using npm just to confirm?

**Comment by scarf005** [comment]
would it be okay to just change to npx? or should i change the whole workflow to use npm?

**Comment by felipecrs** [comment]
> would it be okay to just change to npx? or should i change the whole workflow to use npm?

It would work if it were yarn, but I'm not sure for pnpm. That's because, AFAIK, pnpm builds the node_modules folder by using symlinks instead of actual files.

**Comment by felipecrs** [comment]
We have found a similar issue earlier:

- https://github.com/felipecrs/semantic-release-vsce/issues/295

**Comment by scarf005** [comment]
oh, maybe it could be relevent,  on pnpm vsce must be ran with `--no-dependencies`
- https://github.com/microsoft/vscode-vsce/issues/421#issuecomment-1038911725

**Comment by felipecrs** [comment]
You can try it to disable dependencies, check:

- https://github.com/microsoft/vscode-vsce#configuration:~:text=%22dependencies%22%3A%20true%2C

**Comment by felipecrs** [comment]
I also noticed that your `devDependencies` misses `semantic-release`, it's a peer dependency (required).

**Comment by felipecrs** [comment]
Do you see that when you ran `pnpx vsce verify-pat`, pnpm actually installed the `vsce` package before running it? Ideally it was supposed to be preinstalled (installed as part of `pnpm install`). This is weird.

Nevertheless, try replacing your `pnpx semantic-release` line with:

```bash
PATH="${PWD}/node_modules/.bin:${PATH}" pnpx semantic-release
```

I suspect that it may fix the issue.

**Comment by felipecrs** [comment]
> I also noticed that your `devDependencies` misses `semantic-release`, it's a peer dependency (required).

Try this first though. This would explain why pnpx downloads `semantic-release` before running it.

**Comment by scarf005** [comment]
I tried adding devDependencies, but [it did not work](https://github.com/scarf005/Cataclysm-BN-tools/actions/runs/3256609277/jobs/5347293414). 

will try using the path method.

**Comment by scarf005** [comment]
alright, it still fails but vsce-pat works.
- https://github.com/scarf005/Cataclysm-BN-tools/actions/runs/3256636850/jobs/5347338670

**Comment by scarf005** [comment]
https://github.com/scarf005/Cataclysm-BN-tools/actions/runs/3256670022/jobs/5347399236

I wonder why this error happens:
- vscode:prepublish succeeds with `   out/main.js  639b `
- in package.json main script is defined as `"main": "./out/main.js"` 
- but semantic-release tries to find it in `extensions`

```
Extension entrypoint(s) missing. Make sure these files exist and aren't ignored by '.vscodeignore':
  extension/out/main.js
```

**Comment by scarf005** [comment]
oh nevermind, it indeed was `.vscodeignore` problem.
thanks for your support!

**Comment by felipecrs** [comment]
Just to summarize, you had to put PATH in the command line. Right?

**Comment by scarf005** [comment]
wait, for some reason this workflow file worked without setting up PATH: https://github.com/scarf005/Cataclysm-BN-tools/actions/runs/3258033799/workflow

some differences i could think of is that
1. I've ran the command `semantic-release` as `package.json` script instead of `pnpx`. It may have fixed `node_modules`.
2. I've set `"private": true` in `package.json`. it might have affected `semantic-release-vsce`.

**Comment by felipecrs** [comment]
I think it's the first measure which actually fixed the issue.

**Comment by felipecrs** [comment]
pnpx is deprecated, I think it can be changed to `pnpm exec`:

https://pnpm.io/6.x/pnpx-cli

I suspect it would work this way.

**Comment by felipecrs** [comment]
Hey @scarf005, I published [v5.2.2](https://github.com/felipecrs/semantic-release-vsce/releases/tag/v5.2.2) which contains a potential fix for this issue. Would you mind testing it? Thanks in advance.
